core/src/test/scala/pureconfig/ApiSuite.scala

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package pureconfig
+
+import java.nio.file.Path
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.config.ConfigFactory
+import org.scalatest._
+import org.scalatest.prop.PropertyChecks
+import pureconfig.PathUtils._
+
+import scala.concurrent.duration.FiniteDuration
+
+class ApiSuite extends FlatSpec with Matchers with OptionValues with EitherValues with PropertyChecks {
+
+  behavior of "pureconfig"
+
+  it should "loadConfig from reference.conf" in {
+    case class Conf(d: Double, i: Int, s: String)
+    loadConfig[Conf] shouldBe Right(Conf(0D, 0, "default"))
+  }
+
+  it should "loadConfig from reference.conf with a namespace" in {
+    case class Conf(f: Float)
+    loadConfig[Conf](namespace = "foo") shouldBe Right(Conf(3.0F))
+  }
+
+  it should "loadConfig from a typelevel Config" in {
+    case class Conf(d: Double, i: Int)
+    val conf = ConfigFactory.parseString("{ d: 0.5, i: 10 }")
+    loadConfig[Conf](conf = conf) shouldBe Right(Conf(0.5D, 10))
+  }
+
+  it should "loadConfig from a Typesafe Config with a namespace" in {
+    case class Conf(f: Float)
+    val conf = ConfigFactory.parseString("foo.bar { f: 1.0 }")
+    loadConfig[Conf](conf = conf, namespace = "foo.bar") shouldBe Right(Conf(1.0F))
+  }
+
+  it should "loadConfig from a configuration file" in {
+    case class Conf(s: String, b: Boolean)
+    val path = createTempFile("""{ b: true, s: "str" }""")
+    loadConfig[Conf](path = path) shouldBe Right(Conf("str", true))
+  }
+
+  it should "loadConfig from a configuration file with a namespace" in {
+    case class Conf(s: String, b: Boolean)
+    val path = createTempFile("""foo.bar { b: true, s: "str" }""")
+    loadConfig[Conf](path = path, namespace = "foo.bar") shouldBe Right(Conf("str", true))
+  }
+
+  it should s"be able to load a realistic configuration file" in {
+    case class DriverConf(cores: Int, maxResultSize: String, memory: String)
+    case class ExecutorConf(memory: String, extraJavaOptions: String)
+    case class SparkAppConf(name: String)
+    case class SparkLocalConf(dir: String)
+    case class SparkNetwork(timeout: FiniteDuration)
+    case class SparkConf(master: String, app: SparkAppConf, local: SparkLocalConf, driver: DriverConf, executor: ExecutorConf, extraListeners: Seq[String], network: SparkNetwork)
+    case class SparkRootConf(spark: SparkConf)
+    val configFile = createTempFile("""
```

Comment:
Suggestion: Drop the `"""` to the line after the `(`so that there isn't so much indentation to the `|`.